src/OrleansRuntime/Streams/QueueBalancer/StreamQueueBalancerFactory.cs

```diff
@@ -21,7 +22,7 @@ internal class StreamQueueBalancerFactory
         /// <param name="siloMaturityPeriod">Maturity Period of a silo for queue rebalancing purposes</param>
         /// <returns>Constructed stream queue balancer</returns>
         public static IStreamQueueBalancer Create(
```

Comment:
This is the internal factory class which is always there, not the public factory interface. 

I kept it because our built in queue balancer need to be configured with dependency which is not in DI or in the queueBalancer.init method param, such as `IDeploymentConfiguration `, `IConsistentRingProviderForGrains`. 

We can remove this class by adding `IDeploymentConfiguration `, `IConsistentRingProviderForGrains` into DI. But adding IDeploymentConfiguration to DI is non-trivia. One of its concrete class is ServiceRuntimeWrapper, which is in AzureUtils. I'm not familiar with how to inject a service to DI in an azure silo